### PR TITLE
Fixes #2616 bind method not found

### LINKED ISSUE #2616: automation: bind method not found
With a recent build I run into this errors:

```text
2016-12-07 15:06:22,113 | ERROR | pool-8-thread-1  | core                             | 120 - org.eclipse.smarthome.automation.core - 0.9.0.201612071256 | [org.eclipse.smarthome.automation.ruleregistry(109)] bind method [setTemplateRegistry] not found; Component will fail
2016-12-07 15:06:22,119 | ERROR | pool-8-thread-1  | core                             | 120 - org.eclipse.smarthome.automation.core - 0.9.0.201612071256 | [org.eclipse.smarthome.automation.ruleregistry(109)] bind method [setTemplateRegistry] not found
```

The class `RuleRegistryImpl` has a `setTemplateRegistry` method using type: `RuleTemplateRegistry`
But the component XML is using type: `org.eclipse.smarthome.automation.template.TemplateRegistry`

```text
2016-12-07 15:06:22,757 | ERROR | pool-8-thread-1  | providers                        | 126 - org.eclipse.smarthome.automation.providers - 0.9.0.201612071256 | [org.eclipse.smarthome.automation.provider.AutomationResourceBundlesTracker(126)] bind method [setProvider] not found; Component will fail
2016-12-07 15:06:22,757 | ERROR | pool-8-thread-1  | providers                        | 126 - org.eclipse.smarthome.automation.providers - 0.9.0.201612071256 | [org.eclipse.smarthome.automation.provider.AutomationResourceBundlesTracker(126)] bind method [setProvider] not found
2016-12-07 15:06:22,795 | ERROR | pool-8-thread-1  | providers                        | 126 - org.eclipse.smarthome.automation.providers - 0.9.0.201612071256 | [org.eclipse.smarthome.automation.provider.AutomationResourceBundlesTracker(126)] bind method [setProvider] not found
```

The class `AutomationResourceBundlesTracker` has a `setProvider` method using type: `AbstractResourceBundleProvider`
But the component XML is using type: `org.eclipse.smarthome.core.common.registry.Provider`

### PR DESCRIPTION
**Pull request by danchom**
Signed-off-by: Yordan Mihaylov <j.mihailov@prosyst.bg>

### CODE REVIEW

**Comment by maggu2810** [comment]
Thanks, I will test it.

File: bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java [hunk 1]
```diff
@@ -154,7 +154,7 @@ protected void unsetModuleTypeRegistry(ModuleTypeRegistry moduleTypeRegistry) {
      *
      * @param templateRegistry templateRegistry service.
      */
-    protected void setTemplateRegistry(RuleTemplateRegistry templateRegistry) {
+    protected void setTemplateRegistry(TemplateRegistry<RuleTemplate> templateRegistry) {
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java]: The generic type information is lost at runtime and the XML only declared TemplateRegistry, I am not sure if this is always correct

File: bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AutomationResourceBundlesTracker.java [hunk 2]
```diff
@@ -84,8 +85,15 @@ protected void deactivate(BundleContext bc) {
         rImporter = null;
     }
 
+    @SuppressWarnings({ "rawtypes" })
+    protected void setProvider(Provider provider) {
```
Comment (maggu2810) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AutomationResourceBundlesTracker.java]: As you have choosen 0..n as cardinality wouldn't be "add" be more precise then "set"?

Comment (danchom) [inline bundles/automation/org.eclipse.smarthome.automation.providers/src/main/java/org/eclipse/smarthome/automation/internal/core/provider/AutomationResourceBundlesTracker.java]: Yes, you are right.

File: bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java (hunk 1 above)
Comment (danchom) [inline bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java]: We don't have other type template registries, but anyway I'll add check for this.

**Comment by maggu2810** [comment]
All fine

**Pull request merged**